docs/2.0/docs/pipelines/installation/addingexistinggitlabrepo.mdx

```diff
@@ -0,0 +1,879 @@
+
+{/* We're using an h3 tag here instead of a markdown heading to avoid adding content to the ToC that won't work when switching between tabs */}
+<h3>Pulling the resources into state</h3>
+
+Once you've provisioned the resources in your Azure subscription, you can pull the resources into state using the storage account we just provisioned.
+
+```bash title="name-of-subscription"
+terragrunt run --all --non-interactive --provider-cache --no-stack-generate -- init -migrate-state -force-copy
+```
+
+:::tip
+
+We're adding the `-force-copy` flag here to avoid any issues with OpenTofu waiting for an interactive prompt to copy up local state.
+
+:::
+
+:::note Progress Checklist
+
+<PersistentCheckbox id="run-init-migrate-state-force-copy" label="Pull the resources into state using Terragrunt." />
+
+:::
+
+</TabItem>
+</Tabs>
+
+## Creating `.gruntwork` HCL Configurations
+
+Create [HCL configurations](/2.0/reference/pipelines/configurations-as-code/) in the `.gruntwork` directory in the root of your project to tell Pipelines how you plan to organize your infrastructure, and how you plan to have Pipelines authenticate with your cloud provider(s).
+
+### The `repository` block
+
+The core configuration that you'll want to start with is the `repository` block. This block tells Pipelines which branch has the "live" infrastructure you want provisioned. When you merge IaC to this branch, Pipelines will be triggered to update your infrastructure accordingly.
+
+```hcl title=".gruntwork/repository.hcl"
+repository {
+  deploy_branch_name = "main"
+}
+```
+
+:::note Progress Checklist
+
+<PersistentCheckbox id="create-repository-hcl" label="Create a new `.gruntwork/repository.hcl` file in the root of your project with the contents above." />
+<PersistentCheckbox id="edit-repository-deploy-branch-name" label="Edit the value of `deploy_branch_name` in `.gruntwork/repository.hcl` if necessary." />
+
+:::
+
+### The `environment` block
+
+Next, you'll want to define the environments you want to manage with Pipelines using the [`environment` block](/2.0/reference/pipelines/configurations-as-code/api#environment-block).
+
+For each environment, you'll want to define a [`filter` block](/2.0/reference/pipelines/configurations-as-code/api#filter-block) that tells Pipelines which units are part of that environment. You'll also want to define an [`authentication` block](/2.0/reference/pipelines/configurations-as-code/api#authentication-block) that tells Pipelines how to authenticate with your cloud provider(s) for that environment.
+
+<Tabs>
+<TabItem value="aws" label="AWS" default>
+
+```hcl title=".gruntwork/environment-production.hcl"
+environment "production" {
+  filter {
+    paths = ["prod/*"]
+  }
+
+  authentication {
+    aws_oidc {
+      account_id         = "123456789012"
+      plan_iam_role_arn  = "arn:aws:iam::123456789012:role/pipelines-plan"
+      apply_iam_role_arn = "arn:aws:iam::123456789012:role/pipelines-apply"
+    }
+  }
+}
+```
+
+:::tip
```

Comment:
These tips should probably be put into the AWS tab above and not shown if the user clicks Azure.

Reply:
They already are, I misread this.